Github Action to automate pipeline

kennethrioja opened the pull request:
(the following text has been added to `README.md`)

`.github/workflows/update_json.yaml` allows to run (every day at 1:00AM or when a push to the main branch is done) `src/download_csv.py` which downloads the sheets of a given spreadsheet under `/csv` then `src/csv_to_json.py` takes the csvs in `/csv` and convert them as json and store them in `/json`. The workflow then add, commits and push the changes to main.

Configuration file is `src/config.yaml`:

- `id`: spreadsheet identifier
- `spreadsheet`: list of objects {name, gid, skip, schemaType}, each of them representing a sheet
- `spreadsheet.name`: name given to the csv file when getting fetched by `download_csv.py`
- `spreadsheet.gid`: sheet identifier (Google ID)
- `spreadsheet.skip`: when set to `True`, the `download_csv.py` script will skip the download the `gid` sheet (mainly done for `spreadsheet.name == "keywords"`)
- `spreadsheet.schemaType`: following [schema.org](https://schema.org/), we gave the appropriate `type` to the kind of materials
- `mapping`: list of objects {tess, schema} which acts as a dictionnary, `mapping.tess` is the field in TeSS corresponding to `mapping.schema` in schema.org, they are repsectively key-pair values.
- `mapping.tess`: metadata field known by TeSS
- `mapping.schema`: metadata field known by schema.org

kennethrioja commented:
Note:

For now, I've only enabled 'training_materials_and_curricula' sheet to be downloaded ~~however TeSS sees that some Descriptions are blank (see screenshot) – this is the next task to do and I can work on it.~~, descriptions for all training materials were manually updated by me. So it is ready to get ingested,

dgarijo commented on src/csv_to_json.py:
```diff
@@ -0,0 +1,65 @@
+
+            for row in reader:
+                resource = {
+                    '@context': 'https://schema.org',
```
```suggestion
                    '@context': 'http://schema.org',
```
It is still recommended to go for `http`, even though it will move at some point to `https`

kennethrioja replied:
Good point!

Source https://schema.org/docs/developers.html:

> As noted in the [FAQ](https://schema.org/docs/faq.html#19), the Schema.org URI is sometimes written with 'http', and sometimes with 'https'. Consuming applications are encouraged to accept both representations, and we now generate downloads in both "flavours" as a convenience. In editorial work "on disk" we use the 'https' form. However other sites and applications are likely to be using 'http', especially in RDF representations. Although we encourage a move towards 'https', note in particular that the JSON-LD context file for Schema.org uses 'http', and we are unlikely to change this without a long warning period due to the impact it would have on deployed installations. Feedback on this is welcome, e.g. in this [github issue](https://github.com/schemaorg/schemaorg/issues/2886).

dgarijo reviewed (changes requested):
The proposed workflow looks good to me. Just one minor change about the schema prefix

kennethrioja commented:
Changes I made: 
- I deleted remnant `.csv` files – these will be organised in the `./csv` directory
- I added more info on the `README.md` file as well as in the code.

dgarijo commented:
@kennethrioja I think it looks good. The only thing I would do is that I would still keep a copy of the spreadsheets transformed into JSON-LD in the repository. The reason being that google sheets may change, and you may not know which version was uploaded in the training registry. If there are any errors, it becomes difficult to debug. You can leave them in a `/data` folder or similar.

dgarijo commented:
After discussing with @kennethrioja it's done.
